### PR TITLE
Port autoware vehicle rviz plugin

### PR DESCRIPTION
**Pull request by sgermanserrano**
Port `autoware_vehicle_rviz_plugin` to ROS2
Signed-off-by: Servando German Serrano <servando.german.serrano@linaro.org>

### CODE REVIEW

**Comment by nik-tier4** [comment]
@sgermanserrano @mitsudome-r looks good to me.

File: common/util/rviz_plugins/autoware_vehicle_rviz_plugin/CMakeLists.txt [hunk 1]
```diff
@@ -1,58 +1,42 @@
...
+find_package(Qt5 ${rviz_QT_VERSION} EXACT REQUIRED Core Widgets)
+set(QT_LIBRARIES Qt5::Widgets)
 
 set(CMAKE_AUTOMOC ON)
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 add_definitions(-DQT_NO_KEYWORDS)
 
 ## Declare a C++ library
-add_library(autoware_vehicle_rviz_plugin
+ament_auto_add_library(autoware_vehicle_rviz_plugin
```
Comment (jilaada) [inline common/util/rviz_plugins/autoware_vehicle_rviz_plugin/CMakeLists.txt:23]: i think you might need to add `SHARED` here in order for the library to be exposed to `rviz2`

**Review by jilaada — CHANGES_REQUESTED** [review]
Overall the changes look good. I was able to compile, although there are some compilation warnings:
```
/home/jilada/AutowareArchitectureProposal/src/autoware/autoware.iv/common/util/rviz_plugins/autoware_vehicle_rviz_plugin/src/tools/jsk_overlay_utils.cpp: In member function ‘virtual bool jsk_rviz_plugins::OverlayObject::isTextureReady()’:
/home/jilada/AutowareArchitectureProposal/src/autoware/autoware.iv/common/util/rviz_plugins/autoware_vehicle_rviz_plugin/src/tools/jsk_overlay_utils.cpp:121:64: warning: ‘bool Ogre::SharedPtr<T>::isNull() const [with T = Ogre::Texture]’ is deprecated [-Wdeprecated-declarations]
  121 | bool OverlayObject::isTextureReady() { return !texture_.isNull(); }
      |                                                                ^
In file included from /opt/ros/foxy/opt/rviz_ogre_vendor/include/OGRE/OgreHardwarePixelBuffer.h:35,
                 from /home/jilada/AutowareArchitectureProposal/src/autoware/autoware.iv/common/util/rviz_plugins/autoware_vehicle_rviz_plugin/src/tools/jsk_overlay_utils.hpp:39,
                 from /home/jilada/AutowareArchitectureProposal/src/autoware/autoware.iv/common/util/rviz_plugins/autoware_vehicle_rviz_plugin/src/tools/jsk_overlay_utils.cpp:36:
/opt/ros/foxy/opt/rviz_ogre_vendor/include/OGRE/OgreSharedPtr.h:91:30: note: declared here
   91 |         OGRE_DEPRECATED bool isNull(void) const { return !shared_ptr<T>::operator bool(); }
      |                              ^~~~~~
/home/jilada/AutowareArchitectureProposal/src/autoware/autoware.iv/common/util/rviz_plugins/autoware_vehicle_rviz_plugin/src/tools/jsk_overlay_utils.cpp: In member function ‘virtual bool jsk_rviz_plugins::OverlayObject::updateTextureSize(unsigned int, unsigned int)’:
/home/jilada/AutowareArchitectureProposal/src/autoware/autoware.iv/common/util/rviz_plugins/autoware_vehicle_rviz_plugin/src/tools/jsk_overlay_utils.cpp:154:1: warning: no return statement in function returning
non-void [-Wreturn-type]
  154 | }
      | ^
```

When starting rviz and adding the plugin I get the following error:
Screenshot from 2020-12-09 16-19-13
The same error occurs for the other 3 classes to be loaded. I took a look to see what might be the issue but I was not very successful :(

File: common/util/rviz_plugins/autoware_vehicle_rviz_plugin/src/tools/velocity_history.cpp [hunk 2]
```diff
@@ -137,12 +137,12 @@ void VelocityHistoryDisplay::updateVisualization()
     "BaseWhiteNoLighting", Ogre::ResourceGroupManager::DEFAULT_RESOURCE_GROUP_NAME);
   material->setSceneBlending(Ogre::SBT_TRANSPARENT_ALPHA);
   material->setDepthWriteEnabled(false);
-  ros::Time current_time = ros::Time::now();
+  rclcpp::Time current_time = rclcpp::Clock().now();
```
[the remark this answers is not in the document]
Comment (sgermanserrano) [reply common/util/rviz_plugins/autoware_vehicle_rviz_plugin/src/tools/velocity_history.cpp:140]: the plugin does not inherit from the `Node` class hence the `Clock` creation there

File: common/util/rviz_plugins/autoware_vehicle_rviz_plugin/src/tools/jsk_overlay_utils.cpp [hunk 3]
```diff
@@ -125,11 +124,13 @@ bool OverlayObject::updateTextureSize(unsigned int width, unsigned int height)
 {
   const std::string texture_name = name_ + "Texture";
   if (width == 0) {
-    ROS_WARN("[OverlayObject] width=0 is specified as texture size");
+    RCLCPP_WARN(rclcpp::get_logger("OverlayObject"),
+      "[OverlayObject] width=0 is specified as texture size");
```
[the remark this answers is not in the document]
Comment (sgermanserrano) [reply common/util/rviz_plugins/autoware_vehicle_rviz_plugin/src/tools/jsk_overlay_utils.cpp]: Fixed in the latest commit

File: common/util/rviz_plugins/autoware_vehicle_rviz_plugin/package.xml [hunk 4]
```diff
@@ -4,22 +4,21 @@
...
+  <depend>libqt5-gui</depend>
+  <depend>libqt5-widgets</depend>
+  <depend>rviz_common</depend>
+  <depend>rviz_default_plugins</depend>
+  <depend>rviz_ogre_vendor</depend>
+  <depend>autoware_vehicle_msgs</depend>
 
   <export>
+    <build_type>ament_cmake</build_type>
     <rviz plugin="${prefix}/plugins/plugin_description.xml"/>
```
[the remark this answers is not in the document]
Comment (sgermanserrano) [reply common/util/rviz_plugins/autoware_vehicle_rviz_plugin/package.xml]: Fixed in the latest commit, it doesn't seem to be needed

File: common/util/rviz_plugins/autoware_vehicle_rviz_plugin/CMakeLists.txt (hunk 1 above)
Comment (sgermanserrano) [reply common/util/rviz_plugins/autoware_vehicle_rviz_plugin/CMakeLists.txt]: Fixed

**Comment by sgermanserrano** [comment]
> Overall the changes look good. I was able to compile, although there are some compilation warnings:
> 
> ```
> /home/jilada/AutowareArchitectureProposal/src/autoware/autoware.iv/common/util/rviz_plugins/autoware_vehicle_rviz_plugin/src/tools/jsk_overlay_utils.cpp: In member function ‘virtual bool jsk_rviz_plugins::OverlayObject::isTextureReady()’:
> /home/jilada/AutowareArchitectureProposal/src/autoware/autoware.iv/common/util/rviz_plugins/autoware_vehicle_rviz_plugin/src/tools/jsk_overlay_utils.cpp:121:64: warning: ‘bool Ogre::SharedPtr<T>::isNull() const [with T = Ogre::Texture]’ is deprecated [-Wdeprecated-declarations]
>   121 | bool OverlayObject::isTextureReady() { return !texture_.isNull(); }
>       |                                                                ^
> In file included from /opt/ros/foxy/opt/rviz_ogre_vendor/include/OGRE/OgreHardwarePixelBuffer.h:35,
>                  from /home/jilada/AutowareArchitectureProposal/src/autoware/autoware.iv/common/util/rviz_plugins/autoware_vehicle_rviz_plugin/src/tools/jsk_overlay_utils.hpp:39,
>                  from /home/jilada/AutowareArchitectureProposal/src/autoware/autoware.iv/common/util/rviz_plugins/autoware_vehicle_rviz_plugin/src/tools/jsk_overlay_utils.cpp:36:
> /opt/ros/foxy/opt/rviz_ogre_vendor/include/OGRE/OgreSharedPtr.h:91:30: note: declared here
>    91 |         OGRE_DEPRECATED bool isNull(void) const { return !shared_ptr<T>::operator bool(); }
>       |                              ^~~~~~
> /home/jilada/AutowareArchitectureProposal/src/autoware/autoware.iv/common/util/rviz_plugins/autoware_vehicle_rviz_plugin/src/tools/jsk_overlay_utils.cpp: In member function ‘virtual bool jsk_rviz_plugins::OverlayObject::updateTextureSize(unsigned int, unsigned int)’:
> /home/jilada/AutowareArchitectureProposal/src/autoware/autoware.iv/common/util/rviz_plugins/autoware_vehicle_rviz_plugin/src/tools/jsk_overlay_utils.cpp:154:1: warning: no return statement in function returning
> non-void [-Wreturn-type]
>   154 | }
>       | ^
> ```
> 
> When starting rviz and adding the plugin I get the following error:
> Screenshot from 2020-12-09 16-19-13
> The same error occurs for the other 3 classes to be loaded. I took a look to see what might be the issue but I was not very successful :( I also noticed during my testing that other plugin packages also don't run as expected :( Some of the requests i put below may not help with getting it to work. So feel free to ignore.

The warnings have been addressed and the plugins can be loaded into `rviz` now

**Review by jilaada — APPROVED** [review]
Was able to see all of the plugins loaded properly. Warning about no description provided but I think that should be addressed in a separate PR as needed. LGTM, thanks for making the fixes! :+1:

**Pull request merged**